diff --git a/node_modules/react-native-record-screen/android/src/main/java/com/recordscreen/RecordScreenModule.kt b/node_modules/react-native-record-screen/android/src/main/java/com/recordscreen/RecordScreenModule.kt
index e3850d5..54a834e 100644
--- a/node_modules/react-native-record-screen/android/src/main/java/com/recordscreen/RecordScreenModule.kt
+++ b/node_modules/react-native-record-screen/android/src/main/java/com/recordscreen/RecordScreenModule.kt
@@ -101,7 +101,8 @@ class RecordScreenModule(reactContext: ReactApplicationContext) : ReactContextBa
   private fun startRecordingScreen() {
     val mediaProjectionManager = reactApplicationContext.getSystemService(Context.MEDIA_PROJECTION_SERVICE) as MediaProjectionManager;
     val permissionIntent = mediaProjectionManager.createScreenCaptureIntent();
-    currentActivity!!.startActivityForResult(permissionIntent, SCREEN_RECORD_REQUEST_CODE);
+    val activity = reactApplicationContext.currentActivity
+    activity!!.startActivityForResult(permissionIntent, SCREEN_RECORD_REQUEST_CODE);
   }
 
 
diff --git a/node_modules/react-native-record-screen/ios/RecordScreen.mm b/node_modules/react-native-record-screen/ios/RecordScreen.mm
index 67ac51f..65ceb5c 100644
--- a/node_modules/react-native-record-screen/ios/RecordScreen.mm
+++ b/node_modules/react-native-record-screen/ios/RecordScreen.mm
@@ -289,13 +289,4 @@ - (int) adjustMultipleOf2:(int)value;
     resolve(@"cleaned");
 }
 
-// Don't compile this code when we build for the old architecture.
-#ifdef RCT_NEW_ARCH_ENABLED
-- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
-    (const facebook::react::ObjCTurboModule::InitParams &)params
-{
-    return std::make_shared<facebook::react::NativeRecordScreenSpecJSI>(params);
-}
-#endif
-
 @end
